### ISSUE TITLE
Can't use JSXExpressionContainer as a child

### ISSUE THREAD

**Issue by mwiencek** [AlexGilleran/jsx-control-statements#16]
I'm using babel 6.2.1 with version 2.0.1 of the plugin. I'd like to do something along the lines of:

```JavaScript
<If condition={foo}>
  {'foo'}
</If>
```

Or with any expression that can be enclosed in `{}`. But I get a nasty error instead:

```
Property consequent of ConditionalExpression expected node to be of a type ["Expression"] but instead got "JSXExpressionContainer" at Object.validate
```

**Comment by AlexGilleran** [comment]
So the _fact_ that there is an error in general makes sense, keep in mind that that desugars to:

`foo ? {'foo'} : ''`

Which isn't valid Javascript or JSX - it has to be either:

```
<If condition={foo}>
  'foo'
</If>
```

or 

```
<If condition={foo}>
 <span> 'foo'</span>
</If>
```

However, our transformer should really catch it and throw a nicer error message than that before it gets to Babel.

**Comment by AlexGilleran** [comment]
(edited the comment above with a code fix and some other thoughts)

**Comment by mwiencek** [comment]
Ah, I see. I was sort of expecting it to desugar into something like `{foo ? 'foo' : undefined}`.

I did try it without the `{}` too, but it throws a `Control statement must have a child!` error, with a bare string.

Another problem with removing the `{}` is that I can't then use a function to return a string/array of elements/etc., which is my main use case.

**Comment by mwiencek** [comment]
e.g. we wrap all of our user-visible strings for gettext lookup/extraction, so the previous example is more accurately:

```JavaScript
<If condition={foo}>
  {l('foo')}
</If>
```

**Comment by AlexGilleran** [comment]
Ah yeah I see how that makes sense.

You can work around it for now by wrapping it in `<span>` - I'll look into allowing `{}` expressions in the next release.

**Comment by mwiencek** [comment]
Wonderful, thanks! :)

**Comment by vkbansal** [comment]
I too am facing a similar issue. I've put `span` tags as work-around for now.
But `ul > span > li` does not make any sense.

**Comment by AlexGilleran** [comment]
Fixed by @texttechne in 3.0.0 :)
